src/connectivity.jl

```diff
@@ -283,6 +283,204 @@ function strongly_connected_components end
+ [2, 3, 1]
+
+
+julia> g=SimpleDiGraph(3)
+{3, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3),(3,2)]
+3-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)
+ (2, 3)
+ (3, 2)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+
+julia> g=SimpleDiGraph(11)
+{11, 0} directed simple Int64 graph
+
+julia> edge_list=[(1,2),(2,3),(3,4),(4,1),(3,5),(5,6),(6,7),(7,5),(5,8),(8,9),(9,8),(10,11),(11,10)]
+13-element Array{Tuple{Int64,Int64},1}:
+ (1, 2)  
+ (2, 3)  
+ (3, 4)  
+ (4, 1)  
+ (3, 5)  
+ (5, 6)  
+ (6, 7)  
+ (7, 5)  
+ (5, 8)  
+ (8, 9)  
+ (9, 8)  
+ (10, 11)
+ (11, 10)
+
+julia> for e in edge_list
+           add_edge!(g, e[1], e[2])
+       end
+
+julia> strongly_connected_components_kosaraju(g)
+4-element Array{Array{Int64,1},1}:
+ [11, 10]    
+ [2, 3, 4, 1]
+ [6, 7, 5]   
+ [9, 8]      
+
+```
+"""
+
+function strongly_connected_components_kosaraju end
+@traitfn function strongly_connected_components_kosaraju(g::AG::IsDirected) where {T, AG <: AbstractGraph{T}}
+       
+   nvg = nv(g)    
+
+   components = Vector{Vector{T}}()    # Maintains a list of strongly connected components
+   sizehint!(components, nvg)
+   
+   order = Vector{T}()         # Vector which will store the order in which vertices are visited
+   sizehint!(order, nvg)
+   
+   color = zeros(UInt8, nvg)       # Vector used as for marking the colors during dfs
+   
+   # dfs1
+   for v in vertices(g)
+       
+       color[v] != 0  && continue  
+       color[v] = 1
+       
+       # Start dfs from v
+       dfs_stack = Vector{T}([v])   # Stack used for dfs. Also push v to the stack
+       
+       while !isempty(dfs_stack)
+           u = dfs_stack[end]
+           w = 0
+       
+           for n in outneighbors(g, u)
+               if  color[n] == 0
+                   w = n
+                   break
+               end
+           end
+           
+           if w != 0
+               push!(dfs_stack, w)
+               color[w] = 1
+           else
+               push!(order, u)  #Push back in vector to store the order in which the traversal finishes(Reverse Topological Sort)
+               color[u] = 2
+               pop!(dfs_stack)    
+           end
+       end
+   end
+    
+   reverse!(g)   # Reverse the graph (Transpose of the graph) 
+    
+   for i = 1:nvg
+        color[i] = 0    # Marking all the vertices from 1 to n as unvisited for dfs2
+   end
+   
+   # dfs2
+   for i in 1:nvg
+       
+       v = order[nvg-i+1]   # Reading the order vector in the decreasing order of finish time
+       color[v] != 0  && continue  
+       color[v] = 1
+       
+       component=Vector{T}()   # Vector used to store the vertices of one component temporarily
+       
+       # Start dfs from v
+       dfs_stack = Vector{T}([v])   # Stack used for dfs. Also push v to the stack
+       
+       while !isempty(dfs_stack)
+           u = dfs_stack[end]
+           w = 0
+       
+           for n in outneighbors(g, u)
```

Comment:
If you would replace `outneighbors` by `inneighbors` here, maybe it would be possible to avoid having to reverse the graph?

Reply:
That was really smart @simonschoelly  _/\_. Using `inneighbors` works well and I don't know why I didn't think of this.